src/H5ESint.c

```diff
@@ -817,14 +820,21 @@ H5ES__get_err_info_cb(H5ES_event_t *ev, void *_ctx)
     /* The 'app_func_name', 'app_file_name', and 'api_name' strings are statically allocated (by the compiler)
      * so there's no need to duplicate them internally, but they are duplicated
      * here, when they are given back to the user.
+     *
+     * They are also allowed to be NULL, so only check for NULL values from
+     * the strdup when the original string is non-NULL.
      */
-    if (NULL == (ctx->curr_err_info->api_name = H5MM_strdup(ev->op_info.api_name)))
+    ctx->curr_err_info->api_name = H5MM_strdup(ev->op_info.api_name);
+    if (ev->op_info.api_name && NULL == ctx->curr_err_info->api_name)
```

Comment:
I'm just curious why using mixed styles, why not:
if (ev->op_info.api_name && !(ctx->curr_err_info->api_name))
or
if (NULL != ev->op_info.api_name && NULL == ctx->curr_err_info->api_name)?